x/bank/keeper/keeper.go

```diff
@@ -32,7 +32,7 @@ type Keeper interface {
 	IterateTotalSupply(ctx sdk.Context, cb func(sdk.Coin) bool)
 	GetSupplyOffset(ctx sdk.Context, denom string) sdk.Int
 	AddSupplyOffset(ctx sdk.Context, denom string, offsetAmount sdk.Int)
-	GetSupplyWithOffset(ctx sdk.Context, denom string) sdk.Coin
```

Comment:
Why do we need this change? I think we might want to maintain compatibility with APIs in the upstream

Reply:
The problem is we need to be able to return a negative number and we can only do this with an int, otherwise we cannot compare the negative diff if we use Coin because Coin can be at a minimum 0

Reply:
Why would supply with offset ever be negative? In my opinion, supply should not be negative by definition

The only reason we have the offset is because we over minted by 225m OSMO at genesis. So we are offsetting these 225m with the offset. Therefore, supply with offset should always be >= 0. If it's negative, there is probably a problem / misconfigiration somewhere

Reply:
Ahh I see - so the claim is that it can be negative in theory so we would like to allow for that?

What is the use case for capturing this? Why would we need to capture by how much it is negative? In Osmosis `x/mint` with mainnet parameters supply with offset should never be negative

Reply:
I have no idea, [this PR](https://github.com/osmosis-labs/cosmos-sdk/pull/295) introduced tests and logic for negative supply due to offset, and when brought into osmosis causes test failures.

Reply:
If [the failing test](https://github.com/osmosis-labs/osmosis/blob/main/x/mint/keeper/keeper_test.go#L354) is the problem, I think we should be refactoring this test instead of the APIs here

Reply:
An example would be [this test](https://github.com/osmosis-labs/osmosis/blob/main/x/mint/keeper/keeper_test.go#L354) in osmosis. Without knowing how negative it went, you just need to assume 0 = the correct negative number

Reply:
So should I just refactor the test to return 0 on an expected negative supply result? I can do this it just seemed...wrong

Reply:
If that's the only occurrence where this matters, I think we should instead refactor that test and pre-mint 225 million + 1, then `SetInitialSupplyOffsetDuringMigration` (-225) and, finally test that the supply with offset == 1 is true.

The reason is because breaking the API compatibility between our fork and the upstream SDK is more difficult to maintain and has higher overhead than adjusting that test.

Sorry I didn't realize this earlier. Please let me know what you think.

Reply:
This makes sense! I will close this PR and modify the test osmosis side. Thank you!

Reply:
Actually, I just checked that supply with offsets have not been upsteamed.

They only exist on our fork so the compatibility limitation does not exist.

Reply:
@czarcas7ic given the above, I think it is more flexible in terms of how we approach this. However, I still do not understand when supply with offset could be negative semantically. From my understanding, if that ever happens, I think we should panic and halt the chain.

Therefore, I'm not sure if allowing for it to be negative via APIs is what we want.